js_modules/dagit/packages/core/src/gantt/toGraphQueryItems.tsx

```diff
@@ -38,9 +38,13 @@ export const toGraphQueryItems = (
         i.dependsOn.every((s) => IStepState.SUCCEEDED === runtimeStepMetadata[s.key]?.state),
       );
       if (!invocationsHappened && keys.length === 0) {
-        keys = [step.key];
+        keyExpansionMap[step.key] = [step.key];
+        continue;
       }
-      keyExpansionMap[step.key] = keys;
+      // The order matters here: add the planned dynamic step at the end, so when displaying the gantt
+      // chart, we can ignore planned dynamic step if any of its invocation exists (i.e. hide the
+      // unresolved node if any resolved node exists).
+      keyExpansionMap[step.key] = [...keys, step.key];
```

Comment:
Ahh I like this! Putting at the end of this list is a bit magical but the comment explains it well and it's a nice lightweight way of solving the display issue.